Modules/+Sources/Msquared.m

```diff
@@ -0,0 +1,481 @@
+                obj.output_monitor = 0;
+            else
+                reply = obj.com('status', 'solstis');
+                
+                obj.etalon_lock =       strcmp(reply.etalon_lock,'on');
+                obj.etalon_voltage =    reply.etalon_voltage;
+%                 obj.resonator_lock =    strcmp(reply.cavity_lock,'on');
+                obj.resonator_voltage = reply.resonator_voltage;
+                obj.output_monitor =    reply.output_monitor;
+            end
+            
+            obj.callGetWavelength()
+        end
+        
+        function getWavemeterWavelength(obj)        % Measures the output wavelength directly from the wavemeter. Also uses the EMM channel if the EMM module is active.
+            if isempty(obj.hwserver) || ~isvalid(obj.hwserver)
+                obj.NIR_wavelength =  NaN;
+                obj.VIS_wavelength =  NaN;
+                obj.diff_wavelength = 1950;
+            else
+                obj.NIR_wavelength = obj.hwserver.com('wavemeter', 'GetWavelengthNum', obj.NIR_channel, 0);
+
+                if strcmp(obj.determineModule(obj.setpoint_), obj.moduleVIS)
+                    obj.hwserver.com('wavemeter', 'SetSwitcherSignalStates', obj.VIS_channel, 1, 1);
+                    obj.VIS_wavelength = obj.hwserver.com('wavemeter', 'GetWavelengthNum', obj.VIS_channel, 0);
+
+                    if obj.VIS_wavelength > 0 && obj.NIR_wavelength > 0
+                        diff_wavelength_ = 1/(1/obj.VIS_wavelength - 1/obj.NIR_wavelength);
+
+                        if diff_wavelength_ < 1951 && diff_wavelength_ > 1949   % This range is just so a silly value isn't set. Not sure if this is a good range.
+                            obj.diff_wavelength = diff_wavelength_;
+                        end
+                    end
+                else
+                    obj.hwserver.com('wavemeter', 'SetSwitcherSignalStates', obj.VIS_channel, 0, 0);
+                    obj.VIS_wavelength = NaN;
+                end
+            end
+        end
+    end
+    
+    methods     % The meat of this tunable source.
+        function tune(obj, target)                  % This is the tuning method that interacts with hardware (target in nm)
+            if isnan(target); return; end           % Do nothing if NaN
+            
+            assert(~isempty(obj.hwserver) && isobject(obj.hwserver) && isvalid(obj.hwserver), 'No hwserver host!')
+            
+            module = obj.determineModule(target);
+            
+            if isempty(module)
+                error(['Wavelength ' num2str(target) ' nm is outside the range of this msquared laser.'])
+            end
+            
+            obj.updatingVal = true;
+            obj.active_module = module;
+
+            if strcmp(module, obj.moduleVIS)
+                obj.emm_setpoint = target;
+                try
+                    out = obj.com('set_wavelength', 'EMM', target, .1); % last arg is timeout
```

Comment:
Should timeout be a pref?

Reply:
one for emm, one for solstis, perhaps?

Reply:
Timeout is more for getting things working, rather than changing functionality. Because it's binary (working or not), I don't think the user needs to touch it.